src/hooks/use-polling.ts

```diff
@@ -0,0 +1,37 @@
+import { convertToUIMessages } from '@/lib/utils/ai';
+import { useEffect } from 'react';
+
+type UsePollingOptions = {
+  id: string;
+  onUpdate: (newData: any) => void;
+  interval?: number;
+};
+
+const usePolling = ({ id, onUpdate, interval = 60000 }: UsePollingOptions) => {
```

Comment:
Feels weird to have a generic `hooks/use-polling.ts` file that has bespoke functionality to the chat API

Would be more re-usable with the API route passed in as a param so this hook was less opinionated